components/script/dom/eventtarget.rs

```diff
@@ -244,11 +244,18 @@ impl CompiledEventListener {
     }
 }
 
-#[derive(Clone, DenyPublicFields, JSTraceable, MallocSizeOf, PartialEq)]
+#[derive(Clone, DenyPublicFields, JSTraceable, MallocSizeOf)]
 /// A listener in a collection of event listeners.
 struct EventListenerEntry {
     phase: ListenerPhase,
     listener: EventListenerType,
+    once: bool,
+}
+
+impl std::cmp::PartialEq for EventListenerEntry {
+    fn eq(&self, other: &Self) -> bool {
+        self.phase == other.phase && self.listener == other.listener
```

Comment:
Why is `once` ignored?

Reply:
Because removing an eventlistener without specifying once: true still needs to find the ones that are once